lib/child_process.js

```diff
@@ -418,13 +464,34 @@ function spawnSync(/*file, args, options*/) {
 
   debug('spawnSync', opts.args, options);
 
+  // Validate the timeout, if present.
+  if (options.timeout != null &&
+      !(typeof options.timeout === 'number' && options.timeout >= 0)) {
+    throw new TypeError('"timeout" must be a non-negative number');
```

Comment:
number or integer? Number would imply that a value such as `1.234` would be sufficient.